pagerduty/data_source_pagerduty_event_orchestrations.go

```diff
@@ -0,0 +1,127 @@
+package pagerduty
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/heimweh/go-pagerduty/pagerduty"
+)
+
+func dataSourcePagerDutyEventOrchestrations() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourcePagerDutyEventOrchestrationsRead,
+
+		Schema: map[string]*schema.Schema{
+			"search": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"event_orchestrations": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"integration": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"id": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"parameters": {
+										Type:     schema.TypeList,
+										Computed: true,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"routing_key": {
+													Type:     schema.TypeString,
+													Computed: true,
+												},
+												"type": {
+													Type:     schema.TypeString,
+													Computed: true,
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourcePagerDutyEventOrchestrationsRead(d *schema.ResourceData, meta interface{}) error {
+	client, err := meta.(*Config).Client()
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] Reading PagerDuty Event Orchestrations")
+
+	searchName := d.Get("search").(string)
+
+	return resource.Retry(5*time.Minute, func() *resource.RetryError {
+		resp, _, err := client.EventOrchestrations.List()
+		if err != nil {
+			return resource.RetryableError(err)
+		}
+
+		var orchestrations []*pagerduty.EventOrchestration
+		re := regexp.MustCompile(searchName)
```

Comment:
Could you please change this by `regexp.Compile` and handle the error caused by invalid regex expression?

Additionally, please cover the regex compilation error handling with a test case.